components/src/dynamo/workflow/builder.py

```
@@ -0,0 +1,226 @@
+        if name in self._inputs:
+            raise WorkflowValidationError(f"duplicate workflow input {name!r}")
+        if not isinstance(spec, ValueSpec):
+            raise WorkflowValidationError("workflow inputs must use ValueSpec")
+        self._inputs[name] = spec
+        return ValueRef.for_input(name, self._owner)
+
+    def input(
+        self,
+        name: str,
+        *,
+        type: str,
```

Ruff (0.16.1)

[error] Function argument `type` is shadowing a Python builtin

(A002)